lib/parsers/composer-parser.ts

```diff
@@ -0,0 +1,93 @@
+import * as _ from 'lodash';
+
+import {
+    DepTree,
+    SystemPackages,
+    PackageRefCount,
+    LockFilePackage,
+    ComposerJsonFile,
+    ComposerDependencies,
+} from '../types';
+
+export class ComposerParser {
+    // After this threshold, a package node in the dep tree won't have expanded dependencies.
+    // This is a cheap protection against combinatorial explosion when there's N packages
+    // that depend on each other (producing N! branches of the dep tree).
+    // The value of 150 was chosen as a lowest one that doesn't break existing tests.
+    // Switching to dependency graph would render this trick obsolete.
+    private static MAX_PACKAGE_REPEATS: number = 150;
+
+    public static getVersion(depObj: ComposerJsonFile | LockFilePackage): string {
+        // check for `version` property. may not exist
+        const versionFound: string = _.get(depObj, 'version', '');
+        // even if found, may be an alias, so check
+        const availableAliases = _.get(depObj, 'extra[\'branch-alias\']', []);
```

Comment:
nit: `"extra['branch-alias']"` should  work. Although, personally, I'd avoid lodash (because it does not allow Typescript to check field names and types)

Reply:
Agree on that. Tried and it's worked. Copied it from source, cos before I've seen that TSLint was shouting when you use double quotes for strings. (only single quotes were allowed)